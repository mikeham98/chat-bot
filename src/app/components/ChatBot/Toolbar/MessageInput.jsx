import React from 'react';
import PropTypes from "prop-types";
import styles from '../../../../themes/components/toolbar.scss';

const MessageInput = ({value, onChange, onBlur}) => {
    return (
        <input
            name={'message'}
            placeholder={'Type a message...'}
            type={'text'}
            value={value}
            onChange={onChange}
            onBlur={onBlur}
            className={styles.messageInputWrapper}
        />
    );
};

MessageInput.propTypes = {
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onBlur: PropTypes.func.isRequired
};

export default React.memo(MessageInput);