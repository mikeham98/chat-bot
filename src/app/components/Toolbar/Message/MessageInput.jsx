import React from 'react';
import styles from '../../../../themes/toolbar.scss';

const MessageInput = ({value, onChange, onBlur}) => {
    return (
        <input
            name={'message'}
            placeholder={'Type a message...'}
            type={'text'}
            value={value}
            onChange={onChange}
            onBlur={onBlur}
            className={styles.messageInputWrapper}
        />
    );
};

export default React.memo(MessageInput);