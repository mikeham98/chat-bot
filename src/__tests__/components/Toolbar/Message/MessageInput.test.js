import React from 'react';
import {shallow} from 'enzyme';
import MessageInput from '../../../../app/components/Toolbar/Message/MessageInput';

describe('MessageInput', () => {
    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<MessageInput {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {
            value: 'This is a message',
            onChange: jest.fn(),
            onBlur: jest.fn(),
        };
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return an input component', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
    describe('functionality', () => {
        describe('props', () => {
            describe('onChange', () => {
                it('should be called onChange of input', () => {
                    const event = {e:{target: {value: 'This is also a message'}}};
                    expect(props.onChange).not.toHaveBeenCalled();
                    wrapper().find('input').first().props().onChange(event);
                    expect(props.onChange).toHaveBeenCalledTimes(1);
                    expect(props.onChange).toHaveBeenCalledWith(event);
                });
            });
            describe('onBlur', () => {
                it('should be called onBlur of input', () => {
                    const event = {e:{target: {value: 'This is a message'}}};
                    expect(props.onBlur).not.toHaveBeenCalled();
                    wrapper().find('input').first().props().onBlur(event);
                    expect(props.onBlur).toHaveBeenCalledTimes(1);
                    expect(props.onBlur).toHaveBeenCalledWith(event);
                });
            });
        });
    });
});