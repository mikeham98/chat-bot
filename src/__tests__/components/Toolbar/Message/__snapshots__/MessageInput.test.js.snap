// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput render should return an input component 1`] = `
<input
  className="messageInputWrapper"
  name="message"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  placeholder="Type a message..."
  type="text"
  value="This is a message"
/>
`;
